addons/l10n_ma/models/res_company.py

```diff
@@ -0,0 +1,22 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from markupsafe import Markup
+
+from odoo import api, fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    l10n_ma_ice = fields.Char("ICE", help='Common Company Identifier', default='')
+
+class BaseDocumentLayout(models.TransientModel):
+    _inherit = 'base.document.layout'
+
+    @api.model
+    def _default_company_details(self):
+        company_details = super()._default_company_details()
+        if self.env.company.country_code == 'MA':
+            company_details += Markup(f"<br> ICE: {self.env.company.l10n_ma_ice}")
+        return company_details
+
+    company_details = fields.Html(related='company_id.company_details', readonly=False, default=_default_company_details)
```

Comment:
A related with a default? :thinking:

Reply:
I actually defined it as it was defined in the base model :smile: 
but, I see your point. We actually defining two sources as default which doesn't make sense.
Done, Got rid of related!